src/core/qgsmaplayer.h

```diff
@@ -76,7 +76,7 @@ enum class QgsMapLayerType SIP_MONKEYPATCH_SCOPEENUM_UNNEST( QgsMapLayer, LayerT
  * Base class for all map layer types.
  * This is the base class for all map layer types (vector, raster).
  */
-class CORE_EXPORT QgsMapLayer : public QObject
+class CORE_EXPORT QgsMapLayer : public QObject, public QgsAbstractTemporal
```

Comment:
I don't think map layers should inherit QgsAbstractTemporal here. Each map layer subclass will have its own way of dealing with time, and a fixed time range may not be applicable for a certain subclass. Even for the layer types which could have a fixed time range, we'll need to expose much more control than is available through QgsAbstractTemporal, so I think for map layers we should use a different temporal class.